File: app/src/main/java/lt/zilinskas/marius/testio/api/PlaygroundApi.java
package lt.zilinskas.marius.testio.api;

import java.util.List;

import io.reactivex.Observable;
import lt.zilinskas.marius.testio.api.entities.Server;
import lt.zilinskas.marius.testio.api.entities.Token;
import lt.zilinskas.marius.testio.api.entities.UserInfo;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface PlaygroundApi {

    @Headers("Content-Type: application/json")
    @POST("/v1/tokens")
    Observable<Token> authorize(@Body UserInfo userInfo);

    @GET("/v1/servers")
    Observable<List<Server>> getServers(@Header("Authorization") String authorization);

}

File: app/src/main/java/com/tesonet/testio/data/remote/PlaygroundApi.kt
package com.tesonet.testio.data.remote

import com.tesonet.testio.data.remote.entity.ApiCredentials
import com.tesonet.testio.data.remote.entity.ApiServer
import com.tesonet.testio.data.remote.entity.ApiToken
import kotlinx.coroutines.Deferred
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface PlaygroundApi {
    companion object {
        const val BASE_URL = "http://playground.tesonet.lt/v1/"
    }

    @POST("tokens")
    fun login(@Body credentials: ApiCredentials): Deferred<ApiToken>

    @GET("servers")
    fun getServers(@Header("Authorization") token: ApiToken): Deferred<List<ApiServer>>
}